azurerm/data_source_azuread_service_principal.go

```diff
@@ -55,47 +57,61 @@ func dataSourceArmActiveDirectoryServicePrincipalRead(d *schema.ResourceData, me
 		}
 
 		servicePrincipal = &app
-	} else {
-		apps, err := client.ListComplete(ctx, "")
+
+	} else if _, ok := d.GetOk("display_name"); ok {
+
+		// use the display_name to find the Azure AD service principal
+		displayName := d.Get("display_name").(string)
+		filter := "displayName eq '" + displayName + "'"
```

Comment:
(as above) could we make this `filter := fmt.Sprintf("displayName eq '%q'", displayName)`

Reply:
Makes sense! Added as requested.